tests/unittests/aggregation_tests.py

```diff
@@ -744,6 +744,57 @@ def test_values_unwind(self):
         self.assertListEqual(values1, expected)
         self.assertListEqual(values2, expected)
 
+    @drop_datasets
+    def test_values_optimized(self):
+        dataset = fo.Dataset()
+        dataset.add_samples(
+            [
+                fo.Sample(filepath="image1.png"),
+                fo.Sample(filepath="image2.png"),
+                fo.Sample(filepath="image3.png"),
+                fo.Sample(filepath="image4.png"),
+                fo.Sample(filepath="image5.png"),
+                fo.Sample(filepath="image6.png"),
+            ]
+        )
+
+        pipeline = dataset.aggregate(fo.Values("id"), _mongo=True)
+        self.assertListEqual(pipeline, [{"$project": {"_id": 1}}])
+
+        pipeline = dataset.aggregate(fo.Values("filepath"), _mongo=True)
+        self.assertListEqual(
+            pipeline,
+            [{"$project": {"filepath": 1, "_id": False}}],
+        )
+
+        pipeline = dataset.aggregate(
+            [fo.Values("id"), fo.Values("filepath")],
+            _mongo=True,
+        )
+        self.assertListEqual(
+            pipeline,
+            [[{"$project": {"_id": 1, "filepath": 1}}]],
+        )
+
+        ids1 = dataset.values("id")
+        _ids1 = dataset.values("_id")
+        filepaths1 = dataset.values("filepath")
+        ids2, _ids2, filepaths2 = dataset.values(["id", "_id", "filepath"])
+
+        self.assertEqual(len(ids1), 6)
+        self.assertEqual(len(_ids1), 6)
+        self.assertEqual(len(filepaths1), 6)
+        self.assertEqual(len(ids2), 6)
+        self.assertEqual(len(_ids2), 6)
+        self.assertEqual(len(filepaths2), 6)
+
+        self.assertIsInstance(ids1[0], str)
+        self.assertIsInstance(_ids1[0], ObjectId)
```

Comment:
Can you include a check that the values line up? I worry removing _id might affect the consistency of the order of results, particularly if values is called on a view with a match stage involving an indexed field.

Reply:
Are you suggesting that changing the last stage of an aggregation pipeline from

```py
{"$project": {"value": ...}}
```

```py
{"$project": {"value": ..., "_id": False}}
```

could change the order of the returned documents? That would be very surprising to me.

Reply:
If you're worried about the interplay of `"_id": False` and previous view stages that leverage indexes, then there would need to be specific tests added to validate that. It wouldn't address your concern to validate the order of the `values()` calls in this new test.

Note that there are existing tests in this module that validate the order of `values()` calls.